pagefaultgames/pokerogue
# [Bug] Fix Poison Heal blocking non-Poison damage

## What are the changes?

This fixes the effects of Poison Heal blocking indirect damage from non-Poison sources (e.g. weather, recoil)

## Why am I doing these changes?



As seen in #2448 Poison Heal is currently blocking more sources of damage than it should.

## What did change?


- `data/ability`: `BlockStatusDamageAbAttr` no longer inherits from `BlockNonDirectDamageAbAttr` and is now applied separately. It also no longer crashes the game if applied while the source does not have a status effect.
- `phases`: `PostTurnStatusEffectPhase` now applies both `BlockStatusDamageAbAttr` and `BlockNonDirectDamageAbAttr` separately.

### Screenshots/Videos


Compare Poison Heal's behavior in this video to the video in #2448 :

https://github.com/pagefaultgames/pokerogue/assets/168692175/e36fad6d-a5d9-4dfb-a135-f92ccaced20d

## How to test the changes?




- Using overrides, start a battle with a poisoned Pokemon with the ability Poison Heal and moves that create indirect sources of damage (e.g. Hail, Wild Charge)
- Verify that only Poison damage is blocked, and the source takes damage from all other indirect sources.

## Checklist
